packages/lesswrong/components/users/DialogueSuggestionsPage.tsx

```diff
@@ -0,0 +1,782 @@
+      },
+      update(cache, { data }) {
+        if (!checkId) {
+          cache.modify({
+            fields: {
+              getUsersDialogueChecks(existingChecksRefs = []) {
+                const newCheckRef = cache.writeFragment({
+                  data: data.upsertUserDialogueCheck,
+                  fragment: gql`
+                    fragment NewCheck on DialogueChecks {
+                      _id
+                      __typename
+                      userId
+                      targetUserId
+                      checked
+                      match
+                    }
+                  `
+                });
+                const newData = [...existingChecksRefs, newCheckRef]
+                return newData;
+              }
+            }
+          });
+        }
+      },
+      optimisticResponse: {
+        upsertUserDialogueCheck: {
+          _id: checkId || randomId(),
+          __typename: 'DialogueCheck',
+          userId: currentUser._id,
+          targetUserId: targetUserId,
+          checked: event.target.checked,
+          match: false 
+        }
+      }
+    })
+    
+    if (response.data.upsertUserDialogueCheck.match) {
+      void handleNewMatchAnonymisedAnalytics()
+      handleConfetti(true);
+      setTimeout(() => handleConfetti(false), 5000); // Reset the variable after 5 so it can be clicked again
+    }
+  }
+
+  const checkboxClass = isChecked 
+    ? (isMatched ? classes.checkboxCheckedMatched : classes.checkboxCheckedNotMatched)
+    : classes.checkbox
+
+  return (
+    <>
+      {showConfetti && <ReactConfetti recycle={false} colors={["#7faf83", "#00000038" ]} />}
+      <FormControlLabel
+        control={ 
+          <Checkbox 
+            classes={{
+              root: checkboxClass,
+              checked: classes.checked,
+            }}
+            onChange={event => updateDatabase(event, targetUserId, setShowConfetti, checkId) } 
+            checked={isChecked}
+          />
+        }
+        label=""
+      />
+    </>
+  );
+};
+
+const isMatched = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
+  return userDialogueChecks.some(check => check.targetUserId === targetUserId && check.match);
+};
+
+const isChecked = (userDialogueChecks: DialogueCheckInfo[], targetUserId: string): boolean => {
+  return userDialogueChecks?.find(check => check.targetUserId === targetUserId)?.checked || false;
+};
+
+type MatchDialogueButtonProps = {
+  isMatched: boolean;
+  targetUserId: string;
+  targetUserDisplayName: string;
+  currentUser: any; // replace with the correct type
+  loadingNewDialogue: boolean;
+  createDialogue: (title: string, participants: string[]) => void;
+  classes: ClassesType;
+};
+
+const MatchDialogueButton: React.FC<MatchDialogueButtonProps> = ({
+  isMatched,
+  targetUserId,
+  targetUserDisplayName,
+  currentUser,
+  loadingNewDialogue,
+  createDialogue,
+  classes,
+}) => {
+  return (
+    <div>
+      {isMatched ? (
```

Comment:
If the other side of the ternary is just returning null, cleaner to do an early return above this:
```
if (!isMatched) return null;

return (
  <div>
    ...
  </div>
);
```